build-advisories/src/Roave/SecurityAdvisories/Version.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version, such as 1.0 or 1.0.0.0 or 2.0.1.3.2
+ */
+final class Version
+{
+    const VALIDITY_MATCHER = '/^(?:\d+\.)*\d+$/';
+
+    /**
+     * @var string
+     */
+    private $versionNumbers;
+
+    /**
+     * @param int[] $versionNumbers
+     */
+    private function __construct(array $versionNumbers)
+    {
+        $this->versionNumbers = $versionNumbers;
+    }
+
+    /**
+     * @param string $version
+     *
+     * @return self
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function fromString($version)
+    {
+        if (! preg_match(self::VALIDITY_MATCHER, $version)) {
+            throw new \InvalidArgumentException(sprintf('Given version "%s" is not a valid version string', $version));
+        }
+
+        return new self(array_values(array_map(
+            function ($versionNumber) {
+                return (int) $versionNumber;
+            },
+            explode('.', $version)
+        )));
```

Comment:
I always forget that intval exists as function. Send a PR?
On Sep 3, 2015 07:31, "Christiaan Baartse" <notifications@github.com> wrote:

> In build-advisories/src/Roave/SecurityAdvisories/Version.php
> <https://github.com/Roave/SecurityAdvisories/pull/13#discussion_r38616554>
> :
>
> > +     * @return self
> > +     *
> > +     * @throws \InvalidArgumentException
> > +     */
> > +    public static function fromString($version)
> > +    {
> > +        if (! preg_match(self::VALIDITY_MATCHER, $version)) {
> > +            throw new \InvalidArgumentException(sprintf('Given version "%s" is not a valid version string', $version));
> > +        }
> > +
> > +        return new self(array_values(array_map(
> > +            function ($versionNumber) {
> > +                return (int) $versionNumber;
> > +            },
> > +            explode('.', $version)
> > +        )));
>
> Just out of interest. Why is this implemented like this and not like so?
>
> return new self(array_map('intval', explode('.', $version));
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Roave/SecurityAdvisories/pull/13/files#r38616554>.
>